### ISSUE TITLE
Surprising interpretation of `null` type in JavaScript type annotations

### ISSUE THREAD

**Issue by jugglinmike** [microsoft/TypeScript#43684]
# Bug Report

### 🔎 Search Terms

"null jsdoc"

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about null from JSDoc

### ⏯ Playground Link

([demo](https://www.typescriptlang.org/play?ssl=4&ssc=1&pln=4&pc=2&useJavaScript=true#code/PQKhAIAEAcEMCdYFtwG8B2BXJAjApvAD5YA2JAvuAB7gjABQAZpugMYAuAlgPbriPgAFFQCUaeuHDw87TPD5UAdO24AxTlTwATQSIDc9ckA))

### 💻 Code

```js
/** @param {number|null} x */
function f (x) {
  return x.toFixed();
}
```

### 🙁 Actual behavior

No error reported.

### 🙂 Expected behavior

Error reported:

>     Property 'toFixed' does not exist on type 'number | null'.
>       Property 'toFixed' does not exist on type 'null'.

Or (with `noImplicitAny` enabled) error reported:

>     Parameter 'x' implicitly has an 'any' type.

---

The documentation includes a summary of the differences between typchecking
`.js` files and `.ts` files. The above behavior seems related to the following
difference:

> ### null, undefined, and empty array initializers are of type any or any[]
>
> Any variable, parameter or property that is initialized with null or
> undefined will have type any, even if strict null checks is turned on. Any
> variable, parameter or property that is initialized with [] will have type
> any[], even if strict null checks is turned on. The only exception is for
> properties that have multiple initializers as described above.

([source](https://www.typescriptlang.org/docs/handbook/type-checking-javascript-files.html#null-undefined-and-empty-array-initializers-are-of-type-any-or-any))

However, it's not clear that the `null` *type* will be interpreted in this way.
If that's what's intended, I'd be a little surprised, but I would also
recommend the following changes:

- updating the documentation to state that this applies even when explicitly typing values as `null`
- updating the `noImplicitAny` rule to warn about this case

**Comment by MartinJohns** [comment]
You need to enable `strictNullChecks`, otherwise there is no `null` type.

**Comment by jugglinmike** [comment]
Ahah, thanks for the tip!
